src/NodeApi/Runtime/NodeEmbedding.cs

```diff
@@ -0,0 +1,420 @@
+    internal static readonly unsafe delegate* unmanaged[Cdecl]<
+        nint, node_embedding_runtime, napi_env, napi_value, napi_value, void>
+        s_runtimePreloadCallback = &RuntimePreloadCallbackAdapter;
+    internal static readonly unsafe delegate* unmanaged[Cdecl]<
+        nint, node_embedding_runtime, napi_env, napi_value, napi_value, napi_value, napi_value>
+        s_runtimeLoadingCallback = &RuntimeLoadingCallbackAdapter;
+    internal static readonly unsafe delegate* unmanaged[Cdecl]<
+        nint, node_embedding_runtime, napi_env, napi_value, void>
+        s_runtimeLoadedCallback = &RuntimeLoadedCallbackAdapter;
+    internal static readonly unsafe delegate* unmanaged[Cdecl]<
+        nint, node_embedding_runtime, napi_env, nint, napi_value, napi_value>
+        s_moduleInitializeCallback = &ModuleInitializeCallbackAdapter;
+    internal static readonly unsafe delegate* unmanaged[Cdecl]<nint, NodeEmbeddingStatus>
+        s_taskRunTaskCallback = &TaskRunCallbackAdapter;
+    internal static readonly unsafe delegate* unmanaged[Cdecl]<
+        nint,
+        node_embedding_task_run_callback,
+        nint,
+        node_embedding_data_release_callback,
+        nint,
+        NodeEmbeddingStatus>
+        s_taskPostCallback = &TaskPostCallbackAdapter;
+    internal static readonly unsafe delegate* unmanaged[Cdecl]<
+        nint, napi_env, void>
+        s_nodeApiRunCallback = &NodeApiRunCallbackAdapter;
+#else
+    internal static readonly node_embedding_data_release_callback.Delegate
+        s_releaseDataCallback = ReleaseDataCallbackAdapter;
+    internal static readonly node_embedding_platform_configure_callback.Delegate
+        s_platformConfigureCallback = PlatformConfigureCallbackAdapter;
+    internal static readonly node_embedding_runtime_configure_callback.Delegate
+        s_runtimeConfigureCallback = RuntimeConfigureCallbackAdapter;
+    internal static readonly node_embedding_runtime_preload_callback.Delegate
+        s_runtimePreloadCallback = RuntimePreloadCallbackAdapter;
+    internal static readonly node_embedding_runtime_loading_callback.Delegate
+        s_runtimeLoadingCallback = RuntimeLoadingCallbackAdapter;
+    internal static readonly node_embedding_runtime_loaded_callback.Delegate
+        s_runtimeLoadedCallback = RuntimeLoadedCallbackAdapter;
+    internal static readonly node_embedding_module_initialize_callback.Delegate
+        s_moduleInitializeCallback = ModuleInitializeCallbackAdapter;
+    internal static readonly node_embedding_task_run_callback.Delegate
+        s_taskRunCallback = TaskRunCallbackAdapter;
+    internal static readonly node_embedding_task_post_callback.Delegate
+        s_taskPostCallback = TaskPostCallbackAdapter;
+    internal static readonly node_embedding_node_api_run_callback.Delegate
+        s_nodeApiRunCallback = NodeApiRunCallbackAdapter;
+#endif
+
+#if UNMANAGED_DELEGATES
+    [UnmanagedCallersOnly(CallConvs = new[] { typeof(CallConvCdecl) })]
```

Comment:
I wonder if we can conditionally define a fake `UnmanagedCallersOnlyAttribute` to eliminate all these `#if` conditions. It's not important for this PR though.

Reply:
It works, but it requires to remove the managed delegates since now the functions are considered to be unmanaged-only by the compiler. When I do it I am starting to see some unexpected test deadlocks. Also, one of the comments say that `The xmldoc markdown tool doesn't support unmanaged delegates`. 
Let's discuss this change for in a future PR.

Reply:
> now the functions are considered to be unmanaged-only by the compiler

I was hoping that a "fake" attribute would not have that affect.

> Let's discuss this change for in a future PR.

Yes, it was just an observation here that is not blocking this PR.

Reply:
Oh, in the case when `UnmanagedCallersOnlyAttribute` is defined but we don't want to use it, we'd have to still define the fake attribute and add a using statement to disambiguate.